### PR TITLE
[WIP]interpreters: Add initial duktape support

### PR DESCRIPTION
**Pull request by no1wudi**
## Summary
Add initial support for duktape js interpreter
## Impact
None
## Testing
REPL and string eval tested on stm32f746g-disco.

### CODE REVIEW

File: interpreters/duktape/.gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+duktape-*
```
Comment (xiaoxiang781216) [inline interpreters/duktape/.gitignore:1]: append \n?

File: interpreters/duktape/Kconfig [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+if INTERPRETERS_DUKTAPE
+
+config INTERPRETERS_DUKTAPE_PRIORITY
+	int "Duktape interpreter priority"
+	default 100
+
+config INTERPRETERS_DUKTAPE_STACKSIZE
+	int "Duktape interpreter stack size"
+	default 8192
```
Comment (xiaoxiang781216) [inline interpreters/duktape/Kconfig:18]: can we use DEFAULT_TASK_STACKSIZE?

File: interpreters/duktape/Makefile [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+# under the License.
+#
+############################################################################
+
+include $(APPDIR)/Make.defs
+
+DUKTAPE_VERSION = 2.5.0
+DUKTAPE_UNPACK = duktape-$(DUKTAPE_VERSION)
+DUKTAPE_TARBALL = duktape-$(DUKTAPE_VERSION).tar.xz
+DUKTAPE_URL	= https://github.com/svaarala/duktape/releases/download/v$(DUKTAPE_VERSION)/$(DUKTAPE_TARBALL)
```
Comment (xiaoxiang781216) [inline interpreters/duktape/Makefile:26]: remove the extra space before =

File: interpreters/duktape/Kconfig (hunk 2 above)
Comment (no1wudi) [reply interpreters/duktape/Kconfig:18]: In my test, the task stack size must greater than 6k to run REPL

File: interpreters/duktape/Makefile (hunk 3 above)
Comment (acassis) [reply interpreters/duktape/Makefile:26]: > remove the extra space before =

yes, or add space in the remaining to align all "=" to the same column, it is also acceptable :-)

**Comment by patacongo** [comment]
was this ready to be merged?  It is marked [WIP]

File: interpreters/duktape/Makefile [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+include $(APPDIR)/Make.defs
+
+DUKTAPE_VERSION = 2.5.0
+DUKTAPE_UNPACK  = duktape-$(DUKTAPE_VERSION)
+DUKTAPE_TARBALL = duktape-$(DUKTAPE_VERSION).tar.xz
+DUKTAPE_URL     = https://github.com/svaarala/duktape/releases/ \
+                    download/v$(DUKTAPE_VERSION)/$(DUKTAPE_TARBALL)
+
+CSRCS = $(DUKTAPE_UNPACK)/src-noline/duktape.c
```
Comment (Ouss4) [inline interpreters/duktape/Makefile:29]: @no1wudi  @xiaoxiang781216 Don't you need to extend `clean::` to delete the object generated by this file?

**Pull request merged**